src/scss/root/_index.scss

```diff
@@ -54,6 +54,18 @@ $tokens: () !default;
 	}
 }
 
+@mixin component-with-states($name, $states: []) {
```

Comment:
This is interesting, but I wonder if it might be confusing or be abused for things that are not "state"

Reply:
I think this is probably fine as is... if anything comes up we can always revisit.